xcms 4: native support for MsExperiment and Spectra

jorainer opened the pull request:
This PR adds native support for `Spectra` to `xcms`: in addition to a `OnDiskMSnExp` it is now possible to run the full `xcms` pre-processing on a `MsExperiment` object that bases/uses `Spectra` for MS data representation.

Also, some internal functions were refactored which should result in some significant speed improvement for some tasks.

jorainer committed:
feat: add first functions to perform peak detection on MsExperiment

jorainer committed:
feat: add findChromPeaks for MsExperiment

- Add functionality for peak detection (and extraction) for `MsExperiment` and
  `XcmsExperiment` objects.

jorainer committed:
feat: finalize peak detection on MsExperiment

- Includes all code to perform peak detection.
- Help pages.
- Unit tests.

jorainer committed:
ci: install MSnbase from github

jorainer committed:
feat: add filterRt,XcmsExperiment

jorainer committed:
feat: add filterFile,MsExperiment

jorainer committed:
feat: add compatibility functions for MSnbase objects

jorainer committed:
refactor: general spectra chunk processing

jorainer committed:
feat: add profMat method for MsExperiment

jorainer committed:
feat: add obiwarp alignment for MxExperiment

jorainer committed:
feat: add adjustRtime,MsExperiment,ObiwarpParam

- Add obiwarp alignment.
- Add `dropAdjustedRtime`.
- Add `applyAdjustedRtime`.
- Modify `dropChromPeaks`.

jorainer committed:
feat: add progress bar for chunk-wise processing

jorainer committed:
feat: add first set of correpondence methods

jorainer committed:
feat: add groupChromPeaks,XcmsExperiment,PeakDensityParam

jorainer committed:
docs: restructure documentation for alignment and correspondence

jorainer committed:
feat: add refineChromPeaks,XcmsExperiment,CleanPeaksParam

jorainer committed:
feat: add first functions for peak refinement on XcmsExperiment

jorainer committed:
ci: fix branch for r-libs action

jorainer committed:
feat: add function for peak merging on XcmsExperiment

jorainer committed:
feat: add refineChromPeaks,XcmsExperiment,MergeNeighboringPeaksParam

jorainer committed:
ci: use R devel and BioC 3.17 for all platforms

jorainer committed:
docs: change role for developers

jorainer committed:
feat: add all refineChromPeaks methods for XcmsExperiment

jorainer committed:
feat: add featureValues function

jorainer committed:
feat: add fillChromPeaks,XcmsExperiment,ChromPeakAreaParam

jorainer committed:
feat: add fillChromPeaks for XcmsExperiment

jorainer committed:
feat: add dropFilledChromPeaks and fix subsetting

- Subsetting of an `XcmsExperiment` considers now also feature definitions
  properly.

jorainer committed:
feat: add import function for MsExperiment

jorainer committed:
refactor: filterRt,XcmsExperiment also subsets features

jorainer committed:
feat: finalize chromPeakSpectra and add readMsExperiment

jorainer committed:
fix: import Hits from S4Vectors

jorainer committed:
fix: import findMatches

jorainer committed:
fix: import findMatches

jorainer committed:
fix: import as.matrix function from S4Vectors

jorainer committed:
fix: call directly S4Vectors::as.matrix

jorainer committed:
fix: import $ from Spectra

jorainer committed:
fix: require Spectra 1.9.4

jorainer committed:
fix: error in subset alignment with obiwarp

- Fix an error in subset alignment with obiwarp for `MsExperiment` objects.

jorainer committed:
feat: add manualChromPeaks,XcmsExperiment

- Add `manualChromPeaks` generic.
- Add `manualChromPeaks` method for `OnDiskMSnExp`, `MsExperiment` and
  `XcmsExperiment`.

jorainer committed:
refactor: use dplyr::between instead of MsCoreUtils::between

jorainer committed:
feat: add filterChromPeaks,XcmsExperiment

jorainer committed:
feat: add manualFeatures,XcmsExperiment

jorainer committed:
feat: add filterFeatureDefinitions,XcmsExperiment

jorainer committed:
Add filterFeatureDefinitions,XcmsExperiment

jorainer committed:
feat: add featureSpectra

jorainer committed:
feat: add base functionality to extrach chromatograms

- Add infrastructure to extract (old) chromatographic data from an
  `MsExperiment`.

jorainer committed:
refactor: import sumi from MsCoreUtils

jorainer committed:
fix: issue #653

jorainer committed:
Merge branch 'master' into spectra

jorainer committed:
refactor: use MsCoreUtils::between and update DIA

- Improve/fix the DIA analysis: avoid merging of data from different isolation
  windows for the overlapping m/z range. Huge performance increase by rewriting
  DIA EIC correlation.
- Use MsCoreUtils::between.

jorainer committed:
Merge branch 'jomain' into spectra

jorainer committed:
tests: add unit tests for extracting chromatographic data

jorainer committed:
feat: add chromatogram,MsExperiment function

jorainer committed:
feat: chromatograms,XcmsExperiment returns also features

- Finalize chromatograms,XcmsExperiment to also return chromatographic peaks and
  Features.

jorainer committed:
feat: add featureChromatograms,XcmsExperiment

jorainer committed:
refactor: use MsCoreUtils::maxi for BPC

jorainer committed:
refactor: import readMsExperiment from MsExperiment (1.1.2)

jorainer committed:
Merge branch 'master' into spectra

jorainer committed:
feat: add processHistory and plotAdjustedRtime

- Add `processHistory` for `XcmsExperiment`.
- Refactor `plotAdjustedRtime` to increase performance.

jorainer committed:
feat: add adjustedRtime,XcmsExperiment

jorainer committed:
feat: add plot,XcmsExperiment

jorainer committed:
feat: add filterMzRange and uniqueMsLevels

jorainer committed:
feat: add filterMzRange

jorainer committed:
feat: support XcmsExperiment for last functions

- Add support for `XcmsExperiment` to `featureSummary` and `overlappingFeatures`.

jorainer committed:
Merge pull request #665 from sneumann/jospectra

feat: support XcmsExperiment for last functions

jorainer committed:
docs: update README

jorainer committed:
ci: force installation of mzR from source

jorainer committed:
ci: force install of mzR from source

jorainer committed:
Merge remote-tracking branch 'origin/spectra' into spectra

jorainer committed:
tests: fix longtests

sneumann committed:
Merge pull request #667 from sneumann/jomain

tests: fix longtests

jorainer committed:
Merge branch 'master' into spectra

jorainer committed:
Merge branch 'spectra' into jospectra

jorainer committed:
docs: update main vignette to use new objects

sneumann commented on R/XcmsExperiment.R:
```diff
@@ -0,0 +1,1707 @@
+#' @title Next Generation `xcms` Result Object
```
Cosmetic hint on framing: "next generation" rarely ages well. Similar to "-final" in word files, where inevitably "-final2" crops up.

jorainer replied:
good point - do you have an alternative for a good title?

sneumann commented on tests/testthat/test_XcmsExperiment-functions.R:
```diff
@@ -0,0 +1,125 @@
+    expect_true(length(res@processHistory) > length(tmp@processHistory))
+    expect_true(sum(is.na(featureValues(res))) < sum(is.na(featureValues(tmp))))
+
+    ## Compare with XCMSnExp.
```
I like that idea to cross-compare results. I guess many of those implicit checks now are being checked more explicitly

sneumann commented:
Hi, I am super-impressed by the amount of work into this branch, and looking forward to use the XcmsExperiment based classes in the future ! Yours, Steffen

sneumann committed:
Move multtest from Suggests to Imports in dependencies

sneumann committed:
Merge pull request #672 from sneumann/fix/issue671

Move multtest from Suggests to Imports in dependencies

jorainer committed:
fix: small updates and fixes for XcmsExperiment

jorainer committed:
Merge branch 'devel' into spectra

jorainer committed:
feat: add loadXcmsData and add xmse test data file

jorainer committed:
feat: add new XcmsExperiment test object

- Add function `loadXcmsData` to load the different test data sets and update
  the path to the original data files.
- Use the new `XcmsExperiment` test data sets for all examples.

jorainer committed:
feat: implement feature grouping functions for XcmsExperiment

jorainer committed:
ci: use R "next"

jorainer committed:
docs: start updating the LC-MS feature grouping vignette

jorainer committed:
Merge branch 'jospectra' into spectra

jorainer committed:
docs: continue updating compounding vignette

jorainer committed:
fix: fix for issue #511

jorainer committed:
Merge branch 'jomain' into spectra

jorainer committed:
docs: update the LC-MS compounding vignette

sneumann committed:
Merge pull request #673 from sneumann/jomain

fix: fix for issue #511

jwokaty committed:
bump x.y.z version to odd y following creation of RELEASE_3_17 branch

jwokaty committed:
bump x.y.z version to even y prior to creation of RELEASE_3_17 branch

jorainer committed:
fix: chromatogram avoiding to call filterRt if not needed

jorainer committed:
feat: add a function to split an MsExperiment by spectra variable

jorainer committed:
feat: add findChromPeaksIsolationWindow for MxExperiment

- Make `findChromPeaksIsolationWindow` a generic.
- Implement `findChromPeaksIsolationWindow` for `MsExperiment`.

jorainer committed:
feat: make reconstructChromPeakSpectra a method

- Make `reconstructChromPeakSpectra` a method.
- Add `reconstructChromPeakSpectra` for `XcmsExperiment`.

jorainer committed:
ci: use R 4.3 and BioC 3.18 devel

jorainer committed:
feat: add filterIsolationWindow for MsExperiment and XcmsExperiment

jorainer committed:
docs: update LC-MS/MS vignette

- Update LC-MS/MS vignette.
- Add deisotoping example to the LC-MS/MS vignette.

jorainer committed:
ci: add installation of MetaboCoreUtils

sneumann committed:
Fix merge conflict in Version number

jorainer committed:
ci: fix branch names to ensure documentation is build

jorainer committed:
ci: update pkgdown template

jorainer committed:
feat: include isolationWindow in extraction of MS2 chromatogram

- Extracting MS2 chromatograms will also consider/use the
  isolationWindowTargetMz of Spectra.

jorainer committed:
feat: add parameter isolationWindowTargetMz to chromatogram

- Update and fix all code for `chromatogram,MsExperiment` and
  `chromatogram,XcmsExperiment`.
- Add documentation.
- Add unit tests.
- Update LC-MS/MS vignette.

jorainer commented:
as agreed I closed this PR and will directly push to devel
